synapse/storage/_base.py

```diff
@@ -1327,10 +1327,15 @@ def _invalidate_state_caches_and_stream(self, txn, room_id, members_changed):
         """
         txn.call_after(self._invalidate_state_caches, room_id, members_changed)
 
-        keys = itertools.chain([room_id], members_changed)
-        self._send_invalidation_to_replication(
-            txn, _CURRENT_STATE_CACHE_NAME, keys,
-        )
+        # We need to be careful that the size of the `members_changed` list
+        # isn't so large that it causes problems sending over replication, so we
+        # send them in chunks.
+        members_changed = list(members_changed)
+        for i in range(0, len(members_changed), 100):
```

Comment:
given that mxids have a maxlen of 255 chars, we can still easily overflow the max line length of 16K